app/controllers/api/v2/app_groups_controller.rb

```diff
@@ -205,11 +205,71 @@ def deactivated_by_cluster_name
     }, status: :ok and return
   end
 
+  def fetch_redact_labels
+    valid, error_response = validate_required_keys(
+      [:app_group_secret])
+    render json: error_response, status: error_response[:code] and return unless valid
+
+    all_labels = {}
+    app_group = AppGroup.find_by(secret_key: params[:app_group_secret])
+    if app_group.blank? || !app_group.available?
+      render json: {
+        success: false,
+        errors: ['AppGroup not found or inactive'],
+        code: 404
+      }, status: :not_found and return
+    end
+    
+    static, jsonPath = accumulate_rules(app_group)
+    all_labels['default'] = {
+      StaticRules: static,
+      JsonPathRules: jsonPath,
+    }
+
+    barito_apps =[]
+    app_group.barito_apps.where(status:"ACTIVE").each do |barito_app|
+      static, jsonPath = accumulate_rules(barito_app)
+      if !static.empty? || !jsonPath.empty?
+        all_labels[barito_app.name] = {
+          StaticRules: static,
+          JsonPathRules: jsonPath,
+        }
+      end
+    end
+
```

Comment:
before return the result, we can store this to cache and set expiration data every 5/10 minutes. [sample](https://github.com/BaritoLog/BaritoMarket/blob/master/app/controllers/api/v2/apps_controller.rb#L26-L27)

Reply:
here is sample for set [cache expiry](https://github.com/BaritoLog/BaritoMarket/blob/master/app/services/redis_cache_listener.rb#L37)

Reply:
or we don't need to expired that, but please delete/ renew the cache whenever any update on redact field. you can follow this [approach](https://github.com/BaritoLog/BaritoMarket/blob/master/app/services/redis_cache_listener.rb#L20-L32)

Reply:
added caching on [this](https://github.com/BaritoLog/BaritoMarket/pull/298/commits/2bf22c26ae2dacc5e5a6609cf72aa740a2afde3d) commit